### PR TITLE
Backport/thud/secondaries and ci improvement

### PR DESCRIPTION
**Pull request by mike-sul**
oe-selftest hasn't been executed yet

### CODE REVIEW

**Comment by lbonn** [comment]
First three patches are not relevant: there is one patch applied and removed and adding warrior as a compatible release on sumo doesn't make sense.

**Comment by mike-sul** [comment]
@lbonn @patrickvacek Please, check it out, just two commits left :)

**Comment by patrickvacek** [comment]
This looks correct to me! @lbonn how hard is it to test this in CI? (Preferably alongside the thud PR for meta-updater.)

**Comment by lbonn** [comment]
Would have to do something similar as https://github.com/advancedtelematic/aktualizr/pull/1262, but for this repo...

As a workaround, you can also try to launch a pipeline against the thud branch, with `PIN_LIST=meta-updater-raspberrypi:fea7dd7865d15e369312192365efe8122017074b`

**Comment by patrickvacek** [comment]
Works with the thud PR: https://main.gitlab.in.here.com/olp/edge/ota/connect/client/meta-updater/pipelines/49717

**Comment by patrickvacek** [comment]
@lbonn Is the rpi64 covered by oe-selftest or does it require manual testing? Either way, this (and https://github.com/advancedtelematic/meta-updater/pull/548) look fine to me and I think we can ask QA to take a look at them if you agree.

**Comment by mike-sul** [comment]
@lbonn @patrickvacek This version enables WiFi on my RPi but I don't see Ethernet interface, trying to figure out why...

**Comment by lbonn** [comment]
> @lbonn Is the rpi64 covered by oe-selftest or does it require manual testing?

It requires manual testing, just as regular rpi does: building the image is not enough, we should at least try to boot it.

**Comment by mike-sul** [comment]
@lbonn @patrickvacek This version fixes the issue with ethernet, so now there are both network interfaces, ethX and wlanX.
Would you like me to test the 64 version ?

**Comment by patrickvacek** [comment]
>  Would you like me to test the 64 version ?

Yes, that would be good!

File: recipes-kernel/linux/linux-raspberrypi_4.19.bb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+LINUX_RPI_BRANCH ?= "rpi-4.19.y"
+
+SRCREV = "b45a32b47b76cb64225d19841c20ee85ea97874d"
+SRC_URI = " \
+    git://github.com/raspberrypi/linux.git;protocol=git;branch=${LINUX_RPI_BRANCH} \
+    "
+
+# linux-raspberrypi.inc adds file(s) to SRC_URI: SRC_URI += "file://rpi-kernel-misc.cfg"
+# so we need to add a path to the file(s) to FILESEXTRAPATHS so it can find it
+YOCTOROOT = "${@os.path.abspath(os.path.join("${TOPDIR}", os.pardir))}"
```
Comment (lbonn) [inline recipes-kernel/linux/linux-raspberrypi_4.19.bb]: I would prefer not adding these files from here but copying what we need from a repo with a working 4.19 version to our own `files` directory.

The two main problems:
- it's brittle, we had a new version that's not explicitly supported at the location where the file lie
- it relies on the relative paths between layers

Comment (lbonn) [reply recipes-kernel/linux/linux-raspberrypi_4.19.bb]: We can also try to advocate for fixing this bug on meta-rasbperrypi:thud directly, in addition to this patch.

**Comment by mike-sul** [comment]
> > Would you like me to test the 64 version ?
> 
> Yes, that would be good!

It does work.

root@raspberrypi3-64:/sysroot/home/root# uname -a
Linux raspberrypi3-64 4.19.44 #1 SMP PREEMPT Tue May 21 14:17:33 UTC 2019 aarch64 GNU/Linux
root@raspberrypi3-64:/sysroot/home/root# ip a | grep -A 2 eth0:
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel qlen 1000
    link/ether 4e:5d:5b:4b:6c:71 brd ff:ff:ff:ff:ff:ff
    inet 192.168.88.17/24 brd 192.168.88.255 scope global dynamic eth0
root@raspberrypi3-64:/sysroot/home/root# ip a | grep -A 2 wlan0:
3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel qlen 1000
    link/ether b8:27:eb:ae:52:c7 brd ff:ff:ff:ff:ff:ff
    inet 192.168.88.18/24 brd 192.168.88.255 scope global dynamic wlan0

File: recipes-kernel/linux/linux-raspberrypi_4.19.bb (hunk 1 above)
Comment (mike-sul) [reply recipes-kernel/linux/linux-raspberrypi_4.19.bb]: actually, 4.19 does not require any additional patches or files (see the original recipe 8c9076e).
yes, the relative path can be potentially an issue.

Comment (lbonn) [reply recipes-kernel/linux/linux-raspberrypi_4.19.bb]: Then what happens if you omit these two lines?

Comment (mike-sul) [reply recipes-kernel/linux/linux-raspberrypi_4.19.bb]: an error saying that `rpi-kernel-misc.cfg` cannot be found.
Anyway, please review, the latest version, it actually overrides SRC_URI so no need to search for  `rpi-kernel-misc.cfg`  which is not required in this case.

**Pull request merged**